R/OsModel.R

```diff
@@ -0,0 +1,127 @@
+#' Title - TODO
+#'
+#' Description - TODO
+#'
+#' @slot stan TODO
+#' @slot templated Logical. Defines whether the OsModel object needs to be completed
+#' @export
+OsModel <- setClass(
+    Class = "OsModel",
+    representation = list(
+        stan = "StanModule",
+        templated = "logical"
+    )
+)
+
+
+
+#' LogLogisticModule
+#'
+#' Log logistic module helper function.
+#' @param functions
+#' @param data
+#' @param parameters
+#' @param transformed_parameters
+#' @param generated_quantities
+#' @param priors
+#' @param inits
+#' @param templated
+#' @export
+LogLogisticModule <- function(functions = "os_functions.stan",
+                              data = "os_data.stan",
+                              parameters = "os_parameters.stan",
+                              transformed_parameters = "os_transformed_parameters.stan",
+                              generated_quantities = "os_generated_quantities.stan",
+                              priors = os_prior(),
+                              inits = list(),
+                              templated = TRUE) {
+    st_mod <- StanModule(
+            functions = functions,
+            data = data,
+            parameters = parameters,
+            transformed_parameters = transformed_parameters,
+            generated_quantities = generated_quantities,
+            priors = priors,
+            inits = inits
+        )
+
+    OsModel(
+        stan = st_mod,
+        templated = templated
+    )
+}
+
+
+
+
+
+
+
+#' Parametrize TemplatedStanOs object with the selected Hazardlink
+#' @param osmod TemplatedStanOs object
+#' @param link HazardLink object
+#' @importFrom stringr str_replace
+#' @importFrom stringr str_remove_all
+#' @importFrom stringr str_replace_all
+#' @export
+setGeneric("parametrize", function(osmod, link) {
+    standardGeneric("parametrize")
+})
+
+
+
+
+
+# To be updated
+setMethod(
+    "parametrize",
+    signature(osmod = "OsModel", link = "HazardLink"),
+    function(osmod, link) {
+        newOS <- osmod@stan
+
+        newOS@stan@functions <- str_replace_all(
+            string = osmod@stan@functions,
+            pattern = "<link_arguments>",
+            replacement = paste0("real ", link@parameters, ",")
+        ) |>
+            paste0("\\n ", link@stan@functions) |>
+            str_replace(
```

Comment:
Perhaps we can use `gsub()` instead of `str_replace()` to avoid the stringr dependency ?